package/libgluonutil/src/libgluonutil.c

```diff
@@ -109,6 +109,33 @@ struct json_object * gluonutil_wrap_and_free_string(char *str) {
 	return ret;
 }
 
+
+struct in6_addr* gluonutil_get_mesh_prefix() {
+	struct json_object *site = gluonutil_load_site_config();
+	if (!site)
+		return NULL;
+
+	struct json_object *node_prefix = NULL;
+	if (!json_object_object_get_ex(site, "node_prefix6", &node_prefix))
+		return NULL;
+
+	char *str_prefix = (char*)json_object_get_string(node_prefix);
+	if (!str_prefix)
+		return NULL;
+
+	// we can do the strchr here because in check_site.lua we have validated the format to be some address::/64
+	char *prefix_addr = strndup(str_prefix, strchr(str_prefix, '/')-str_prefix);
```

Comment:
Just use *strchrnul* instead of *strchr*, no need to rely on site checks (I'd like to make the C code as robust as possible, as we don't have Lua's error()/assert() here).

Reply:
done.